### DISCUSSION TITLE
[Solution] TypeError: natten1dqkrpb() takes 4 positional arguments but 5 were given

### DISCUSSION THREAD

**Discussion by havenpersona** [mir-aidj/all-in-one#20] [General]
It looks like some of you are facing the same issues I encountered. I was able to resolve this by making two changes to the code in ./models/dinat.py:
1. Change line 99 from
`attention_scores = self.nattendqkrpb(query_layer, key_layer, self.rpb, self.kernel_size, self.dilation)`
into
`attention_scores = self.nattendqkrpb(query_layer, key_layer, self.rpb, self.dilation)`
2. Change line 109 from
`context_layer = self.nattendav(attention_probs, value_layer, self.kernel_size, self.dilation)`
into
`context_layer = self.nattendav(attention_probs, value_layer, self.dilation)`

**Comment by tae-jun** [comment]
Hi Haven!

The API of NATTEN was changed at some point.

The issue should be fixed if you use the latest version of NATTEN.

Please try! And feel free to reopen this if it doesn't work. Thanks :)

**Reply by havenpersona** [reply]
Hi Taejun, yeah, the latest version of NATTEN doesn't work. My solution works with natten==0.14.5 though
